### PR TITLE
Proposition to create a complex matrix class

### PR DESCRIPTION
**Pull request by JB-H**
**Please check if the PR fulfills these requirements**

- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

**What kind of change does this PR introduce?**

It is a new feature

**What is the current behavior?**

**What is the new behavior (if this is a feature change)?**
complex matrix handled and used to generate Fortescue matrices

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [x] No

**If yes, please check if the following requirements are fulfilled**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

**Other information**:

### CODE REVIEW

File: math/src/main/java/com/powsybl/math/matrix/ComplexMatrix.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+                int colIndexInCartesian = 2 * j;
+
+                // Before building the complex matrix term, check that the 4x4 cartesian bloc can be transformed into a Complex term
+                double t11 = realMatrix.get(rowIndexInCartesian, colIndexInCartesian);
+                double t12 = realMatrix.get(rowIndexInCartesian, colIndexInCartesian + 1);
+                double t21 = realMatrix.get(rowIndexInCartesian + 1, colIndexInCartesian);
+                double t22 = realMatrix.get(rowIndexInCartesian + 1, colIndexInCartesian + 1);
+
+                double epsilon = 0.00000001;
+                if (FastMath.abs(t11 - t22) > epsilon) {
```
Comment (annetill) [inline math/src/main/java/com/powsybl/math/matrix/ComplexMatrix.java:113]: These two if bloc can be shared, and with the first exception of your method, you can introduce a string variable for exception message.

**Review by annetill — COMMENTED** [review]
Just small comments/corrections.

**Review by geofjamg — APPROVED** [review]

File: math/pom.xml [hunk 2]
```diff
@@ -85,6 +85,14 @@
             <artifactId>slf4j-simple</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
[the remark this answers is not in the document]
Comment (geofjamg) [reply math/pom.xml]: apache commons math is an already existing dependency so this is OK

File: math/src/main/java/com/powsybl/math/matrix/ComplexMatrix.java (hunk 1 above)
Comment (geofjamg) [reply math/src/main/java/com/powsybl/math/matrix/ComplexMatrix.java]: Done

**Review by annetill — APPROVED** [review]
Thanks!

**Pull request merged**